src/precice/exceptions.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <stdexcept>
+
+namespace precice {
+
+class Error : public std::runtime_error {
+public:
+  Error(const std::string &what_arg)
+      : std::runtime_error(what_arg){};
+};
+
+class ActionError : public precice::Error {
+public:
+  ActionError(const std::string &what_arg)
+      : precice::Error(what_arg){};
+};
```

Comment:
Related to my comment above. I think we should put this error into the namespace `precice::action`. This makes it clear that the error should only be thrown from this package.